RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitPhase2WeightsAlgoGPU.cu

```diff
@@ -0,0 +1,35 @@
+#include "DataFormats/EcalDigi/interface/EcalDataFrame_Ph2.h"
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+
+#include "EcalUncalibRecHitPhase2WeightsKernels.h"
+#include "EcalUncalibRecHitPhase2WeightsAlgoGPU.h"
+
+namespace ecal {
+  namespace weights {
+
+    void entryPoint(ecal::DigisCollection<calo::common::DevStoragePolicy> const& digis,
+                    EventOutputDataGPU& eventOutputGPU,
+                    cms::cuda::device::unique_ptr<double[]>& weights_d,
+                    cudaStream_t cudaStream) {
+      unsigned int totalChannels = digis.size;
+      // 64 threads per block best occupancy from Nsight compute profiler
+      unsigned int nchannels_per_block = 64;
+      unsigned int threads_1d = nchannels_per_block;
+      unsigned int blocks_1d = (totalChannels / threads_1d) + 1;
```

Comment:
do you really want to always have more threads than blocks ?
otherwise this should probably be
```suggestion
      unsigned int blocks_1d = (totalChannels + threads_1d - 1) / threads_1d;
```